wp-includes/class-wp-service-workers.php

```diff
@@ -521,6 +384,209 @@ public function register_precached_routes( $routes ) {
 		);
 	}
 
+	/**
+	 * Register scripts which are pre-cached.
+	 *
+	 * @todo Consider storing the script $handles to refer to later by the offline page for what to dequeue.
+	 * @since 0.2
+	 * @param array $handles Script handles.
+	 */
+	public function register_precached_scripts( $handles ) {
```

Comment:
@westonruter For these new method, do you think we should provide functionality, i.e. function or hook, to make it easier for theme developers to register their scripts and styles.  I see confusion around _when_ to do the registration process.  For example, should the assets be enqueued and then registered?

Or better yet, could we _add_ configuration parameter to mark scripts and styles for precaching?

Reply:
Thanks for the link to how you envision it being used.  I knew you were working on an example theme implementation.

>I really like like the idea of a new script/style attribute like “preload”.

>And then the service worker could automatically look for any such styles that have this precache flag to include it, and then this list of styles/scripts would then be used to obtain the list of script/style handles to pass to this method automatically.

Right, it seems to fit into how theme developers enqueue.  At the point where you enqueue, tell us if you want to pre-cache or not. Then the plugin (and later core) handles it for you.